iocore/net/UnixNetVConnection.cc

```diff
@@ -990,6 +993,23 @@ UnixNetVConnection::netActivity(EThread *lthread)
   net_activity(this, lthread);
 }
 
+int
+UnixNetVConnection::startEvent(int /* event ATS_UNUSED */, Event *e)
+{
+  MUTEX_TRY_LOCK(lock, get_NetHandler(e->ethread)->mutex, e->ethread);
+  if (!lock.is_locked()) {
+    e->schedule_in(HRTIME_MSECONDS(net_retry_delay));
+    return EVENT_CONT;
+  }
+  if (!action_.cancelled) {
+    connectUp(e->ethread, NO_FD);
+  } else {
+    lock.release();
```

Comment:
@SolidWallOfCode @bneradt What do you think about release the lock here to make the clang-analyzer happy.

Reply:
Does this make double free?

Reply:
I don't see how the early release helps, nor how it would double free. The lock remembers state so it won't try to release the lock again.

Reply:
Looking at the clang report, I don't understand it. Step 17 notes "Memory is released". That's fine - the `UnixNetVC` is destroyed. But what freed memory is used at step 20? That mutex is in the `NetHandler`, not the `UnixNetVC`. None of that is in the `UnixNetVC`. I seriously wonder if `free(e->ethread)` is confusing the analyzer and it thinks the `ethread` has been freed (which is touched from the lock destructor). I think it's well worth a shot to change that to `this->free(e->ethread);` and see that fixes the analyzer issue (without the `lock.release()`).

Reply:
Ah, this change doesn't make sense. I thought the `mutex` is owned by `UnixNetVC` but it's owned by `NetHandler`, as you pointed. I'll try the `this->free(e->ethread)`.